### PR TITLE
Pass GHC options to make_docs.sh, move flags to stack.yaml, and outpu…

### PR DESCRIPTION
**Pull request by balacij**
…t location of generated docs

Note: `--interleaved-output` can be omitted since it's enabled by default since Stack 1.8

Contributes to: #2525

### CODE REVIEW

**Comment by balacij** [comment]
If we can figure out an optimal thread count for github actions, we can likely speed up builds _a bit_. Passing the GHC flags also passes our current GHCTHREADS option, which will help with this (assuming we get gains from a >2 thread count from the Actions host server).

File: code/stack.yaml [hunk 1]
```diff
@@ -58,6 +58,11 @@ extra-deps:
 - unicode-names-3.2.0.0@sha256:fef7241d93170e26265e84553090253a5e8ee207645d47cf271816f5834d07d2
 - unicode-properties-3.2.0.0@sha256:239766a6ac4322329353f6b9cd546024fd8c5f0235c8e32b3cba2d6bd245a699
 
+# Used only for a basic set of flags for haddock
+build:
+  haddock-deps: false
+  haddock-hyperlink-source: false
```
Comment (JacquesCarette) [inline code/stack.yaml:64]: Why?

**Review by JacquesCarette — APPROVED** [review]
I think most people have found that -j2 speeds things up, and that anything higher doesn't really. It might be worth experiment with.

File: code/stack.yaml (hunk 1 above)
Comment (balacij) [reply code/stack.yaml:64]: These were the original flags used for generating the Haddock docs. `haddock-deps` is a switch for building haddocks for dependencies, and `haddock-hyperlink-source` is for creating the links to the source code. I think we can re-enable that, we'd just need to tweak the settings a little bit for it to go to github.

Comment (JacquesCarette) [reply code/stack.yaml:64]: Right, I was only asking about the hyperlinks. It would be nice to have that.

Comment (balacij) [reply code/stack.yaml:64]: I'm sure we can figure it out. I'll make a ticket.

**Pull request merged**